mne/viz/_3d.py

```diff
@@ -1082,22 +1170,13 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
                           "using:\n\n    $ pip install -U pysurfer" %
                           surfer.__version__)
 
-    if time_unit not in ('s', 'ms'):
-        raise ValueError("time_unit needs to be 's' or 'ms', got %r" %
-                         (time_unit,))
-
     if initial_time is not None and surfer_version > v06:
         kwargs = {'initial_time': initial_time}
         initial_time = None  # don't set it twice
     else:
         kwargs = {}
 
-    if time_label == 'auto':
-        if time_unit == 'ms':
-            time_label = 'time=%0.2f ms'
-        else:
-            def time_label(t):
-                return 'time=%0.2f ms' % (t * 1e3)
```

Comment:
I don't quite understand what was the purpose of this. I changed the logic a bit, but I think it should work the same.